pyrk/th_component.py

```diff
@@ -35,30 +42,80 @@ def __init__(self, name=None,
         :type timer: Timer object
         :param heatgen: is this component a heat generator (fuel)
         :type heatgen: bool
+        :param power_tot: power generated in this component
+        :type power_tot: float
+        :param sph: is this component a spherical component, spherical
+        equations for heatgen, conduction are different,
+        post-processing is different too
+        :type sph: bool
+        :param ri: inner radius of the sph/annular component, ri=0 for sphere
+        :type ri: float
+        :param ro: outer radius of the sph/annular component,
+        ro=radius for sphere
+        :type ro: float
         """
         self.name = name
         self.vol = vol.to('meter**3')
-        validation.validate_ge("vol", vol, 0*units.meter**3)
+        self.mat = mat
         self.k = mat.k
         self.cp = mat.cp
         self.dm = mat.dm
-        self.T0 = T0.to('kelvin')
-        validation.validate_num("T", T0)
+        self.timer = timer
         self.T = units.Quantity(np.zeros(shape=(timer.timesteps(),),
                                          dtype=float), 'kelvin')
         self.T[0] = T0
+        self.T0 = T0
         self.alpha_temp = alpha_temp.to('delta_k/kelvin')
-        self.timer = timer
         self.heatgen = heatgen
         self.power_tot = power_tot
         self.cond = {}
         self.conv = {}
-        self.mass = {}
+        self.adv = {}
+        #self.mass = {}
```

Comment:
In preparation for pulling this PR, I was going through and checking the funcitonality of old input files and found that many of them fail. In particular, by commenting out this line, you cause the add_mass_trans method to fail as soon as it is run. If you would like your add_advection function to supercede the add_mass_trans method, that's fine. However, the function should be removed completely, rather than halfway, and where it appears, corrections should be made to move the old code to the new function name.